core/LSU.hpp

```diff
@@ -319,19 +339,55 @@ namespace olympia
         void handleCacheReadyReq_(const MemoryAccessInfoPtr &memory_access_info_ptr);
         void getAckFromCache_(const MemoryAccessInfoPtr &updated_memory_access_info_ptr);
 
+        // Perform cache read
+        void handleCacheRead_();
         // Retire load/store instruction
         void completeInst_();
 
         // Handle instruction flush in LSU
         void handleFlush_(const FlushCriteria &);
 
+        // Instructions in the replay ready to issue
+        void replayReady_(const InstPtr &);
+
+        // Mark instruction as not ready and schedule replay ready
+        void updateInstReplayReady(const InstPtr &);
+
         ////////////////////////////////////////////////////////////////////////////////
         // Regular Function/Subroutine Call
         ////////////////////////////////////////////////////////////////////////////////
 
+        // Check if instruction is present in the any of the queues
+        bool instPresentInQueues(const InstPtr &inst_ptr);
+
+        // Check if the instruction is present in a specific queue
+        bool instInQueue(LoadStoreIssueQueue &queue,const InstPtr &inst_ptr);
```

Comment:
what do we need both `instPresentInQueues` and `instInQueue`? 

Also, by "queue", do we mean just the issue queue (as opposed to load queue, store queue)? If so, may be change the comment to reflect that.

Reply:
Yes I should rename it, its made sense when I had multiple queues